### ISSUE TITLE
Dashboard issue

### ISSUE THREAD

**Issue by Maofree** [PrestaShop/PrestaShop#15319]
Hi
my customer said me that the statistics on the start page of dashboard report sells only of the order with invoices. but the invoice is a document for finance, if you do not have to declare making the invoices or the client does not require an invoice you do not have to generate an invoice. I believe it depends on the states of order, they are activated only with a certain condition. But if you know how many orders you have made, the actual profit and sales, I think it is a major flaw when it comes to seeing sales and products sold.

I can give you the access of this site if you need it

thanks

**Screenshots**
![2](https://user-images.githubusercontent.com/653230/63968102-2cb23c80-ca9f-11e9-9cbf-c3f83349da96.jpg)

![3](https://user-images.githubusercontent.com/653230/63968202-6daa5100-ca9f-11e9-8688-90a10042e593.png)

**Additional information**
PrestaShop version: 1.7.6.1
PHP version: 7.0

**Comment by Vicdetailer** [comment]
HI
I have discussed this problem with my Maofree programmer, I have seen that we cannot see the reality of the sales made as well as the orders that have been made or the profit margin of our sales because all the orders that have not generated an invoice do not count them and I think that should not be so since it is not always necessary to generate an invoice since that corresponds to the taxation of each country. But if we must have the most accurate information on how many sales we have made and orders or the profit margin that our online store is giving us since it is a critical point to know that we are doing a good job or on the contrary we are carrying a bad strategy of sales, if we are selling with the commercial margin that we earn money or on the contrary if we are losing money because we are selling cheaper. I think it is a very important factor for all online store administrators to have exact control of all orders, products sold and the profit margins that we are having
A greeting and thanks in advance

**Comment by khouloudbelguith** [comment]
Hi @maofree, @Vicdetailer,

About the `dashtrends` module (Dashboard widget), it is necessary to regenerate invoices so the statistics will be displayed, if you don't regenerate an invoice for an order, it will not be taken in consideration for the statistics.
here's a quick solution to generate invoices for all orders.
You need to:
1. Navigate to BO => Shop Parameters => Order Settings => Status Tab
2. Create a new order status( for example "generate invoice")
3. In this order status created
3.1 Check "Allow a customer to download and view PDF versions of his/her invoices."
3.2 Check "Hide this status in all customer orders."
3.3 Save
![image](https://user-images.githubusercontent.com/16067358/64002195-5ef48600-cb01-11e9-8bfa-5a7239f82341.png)

4. In the BO => Orders => Select all orders => Change Order Status with Bulk actions => 
![image](https://user-images.githubusercontent.com/16067358/50694597-9a09ec80-103a-11e9-928d-56053ca76046.png)
![image](https://user-images.githubusercontent.com/16067358/50694616-aa21cc00-103a-11e9-8d28-3aa30c9da989.png)

So, all invoices will be generated for all orders without notifying the customer.
You can make a test before applying this solution.
Thanks!

**Comment by Maofree** [comment]
hi
the problem is that if you create invoices then you have to register them

**Comment by khouloudbelguith** [comment]
@maofree, it is the generating of an invoice for your order. isn't?
Thanks!

**Comment by Maofree** [comment]
not for all orders it is necessary to generate an invoice, only if requested, in the other cases Victor will do a receipt to add in the pack before to send it. In this cases he doesn't use order status with that option active

**Comment by khouloudbelguith** [comment]
@maofree, thanks for this clarifications.
Ping @marionf, @colinegin what do you think?
Is it possible to have the `dashtrends` module with statistics independent of the generation of invoices?
It could be a good improvement for this module.
Thanks!

**Comment by Vicdetailer** [comment]
Hi @khouloudbelguith
Exactly what @maofree says, the dastrends would have to give exact information of the sales made independently if invoices are generated or not, since the invoices in a document for finance but does not affect when giving the real sales data, Orders or profit margin.
We look forward

Greetings

**Comment by hacchus** [comment]
@maofree @Vicdetailer Hi, a month ago I made this pull request #14215 that I think it might help you solve your problem.

**Comment by hitinder** [comment]
Hi @khouloudbelguith 
>About the dashtrends module (Dashboard widget), it is necessary to regenerate invoices so the statistics will be displayed, if you don't regenerate an invoice for an order, it will not be taken in consideration for the statistics.
> here's a quick solution to generate invoices for all orders.

Are you sure about the temp solution you provided above? I am asking this because the staus "**Payment received**" already has "**Allow a customer to download and view PDF versions of his/her invoices.**" checked and still the dashboard data is incorrect.

So I am thinking that regenerating the invoice won't help.

I am using v1.7.5.1 BTW.

Thank you.

**Comment by khouloudbelguith** [comment]
@hitinder For each order created, there are different `order statuses` enables you to easily manage your orders and returns, and keep your customers informed of the evolution of their purchase.
The variously available statuses are visible and editable in the "Statuses" page, under the "Order settings" menu.
In the Statuses page, there are different options:
1. Consider the associated order as validated. If enabled, this status marks all associated orders as "paid", and puts them in this same status.
2. Allow a customer to download and view PDF versions of their invoice. If disabled, you will have to send customers their invoice yourself.
3. Hide this state in all customer orders. This enables you to create internal statuses, for you and your team. Customers will never see this in their order status page.
4. Send an e-mail to customer when his/her order status has changed. When enabled, a drop-down menu appears to let you choose which mail template to use.
5. Attach invoice PDF to email. Send an email to the customer with the invoice in PDF format attached.
6. Attach delivery slip PDF to email. Send an email to the customer with the delivery slip in PDF format attached.
7. Set the order as shipped. Be careful: once an order is set as "shipped", it cannot be set back to the previous status.
8. Set the order as paid. Same here: once an order is set as "paid", it cannot be set back to the previous status.
9. Show delivery PDF. Displays the delivery PDF.

In conclusion, to enable the generations of invoices, in the order status you should have this option "Allow a customer to download and view PDF versions of his/her invoices." checked.
In the dashboard, to get exact stats, we must have two options selected:
1. Consider the associated order as validated.
2. Allow a customer to download and view PDF versions of his/her invoices.
![image](https://user-images.githubusercontent.com/16067358/71879129-a59c6180-312d-11ea-8aa8-aad6d7704e0c.png)
About my first solution, we just regenerate invoices to valid orders.

Thanks!

**Comment by hitinder** [comment]
@khouloudbelguith I understand how the order status works. What I am trying to tell you is that invoice generation is not working or it doesn't work for the `dashtrends ` in the BO the way you explained below.

> About the **dashtrends** module (Dashboard widget), it is necessary to regenerate invoices so the statistics will be displayed, if you don't regenerate an invoice for an order, it will not be taken in consideration for the statistics.

Check this below screenshot from my BO. You can see below that I have both the below options checked. However, in the BO it is still showing the wrong data for me. It doesn't consider discount values.

> Consider the associated order as validated.
> Allow a customer to download and view PDF versions of his/her invoices.

![image](https://user-images.githubusercontent.com/9841229/71880319-16994480-3141-11ea-9f8c-5ebe17ab00a6.png)

**Comment by khouloudbelguith** [comment]
@hitinder, in the BO => Orders => Invoices page The option "Enable invoices" is enabled?
![image](https://user-images.githubusercontent.com/16067358/71882274-6d4c5180-3134-11ea-9290-256e9676e688.png)

Thanks!

**Comment by hitinder** [comment]
@khouloudbelguith yes it is enabled in the BO for me.

![image](https://user-images.githubusercontent.com/9841229/71882598-d2f50980-3145-11ea-85b8-d7d8ef999681.png)

**Comment by khouloudbelguith** [comment]
@hitinder, it is ok in my case with PS1.7.6.2
https://drive.google.com/file/d/1yVfF4DGAe-LxgRHi-ISRBen3k2Jy0jLX/view
Thanks to check & feedback.

**Comment by hitinder** [comment]
@khouloudbelguith I am not saying that `dashtrends` is not working. What is wrong with it is that it's not considering the discount amount.

Example: An item worth $20 and cost $5 in an order with 50% discount should show sales amount as $10 and the profit will be $5. Which is not the case at present. 

If you don't mind, please make a new order, apply a 50% discount to it and show how does it come up on the chart.

**Comment by khouloudbelguith** [comment]
@hitinder, this is another issue.
It is added to our bug roadmap, you need to follow this issue: https://github.com/PrestaShop/PrestaShop/issues/15114

Thanks!

**Comment by hitinder** [comment]
@khouloudbelguith okay got you! I mixed up the two issues.

**Comment by hibatallahAouadni** [comment]
Ping @PrestaShop/product-team friendly reminder!

**Comment by boo-code** [comment]
Reproduces on 9.2, and the mechanism is narrower than the thread assumed: it is not the `dashtrends`
module. `dashtrends` reads `AdminStatsController::getTotalSales()`
(`modules/dashtrends/dashtrends.php:361`), and that query lives in core.

The queries decide what counts as a sale with `os.logable = 1`, but date it by `invoice_date`, which is
only written when an order reaches a status whose `invoice` flag is set. Those are two independent flags
on an order status, so a status that validates an order without invoicing it - which is exactly the
"receipt instead of an invoice" case in the report - produces an order that is a sale by the query's own
test and has no invoice date at all. The date window then excludes it.

Measured on 9.2 with two identical orders placed the same day, one in a status with
`logable = 1, invoice = 1` and one with `logable = 1, invoice = 0`:

```
  orders matching the queries' own validity test (os.logable = 1)   2, worth 3038.40 net
  AdminStatsController::getTotalSales(today, today)                 1519.20
  AdminStatsController::getOrders(today, today)                     1
```

Half the revenue and half the orders, silently. Note no shipped status is logable-without-invoice out of
the box, so this needs a status the merchant created - which is what both reporters describe.

`#36564` covers the same defect and my #42181 fixes it: the date a statistic counts an order at falls
back to the order date when there is no invoice date, so nothing moves for an order that has one. With
that branch applied the same measurement gives 3038.40 and 2.

Two queries in that file were left on the raw column and I have since covered them as well - the
best-selling category and the 31-day average-cart KPI. The first needed the date test qualified with the
orders table's alias, because `getBestCategory()` joins it next to the product table, which carries a
`date_add` of its own; unqualified it fails with an ambiguous column.

The same pattern is in the stats modules, which are separate repositories -
`statssales`, `statsbestproducts`, `statsbestcustomers`, `statsbestcategories`, `statsbestsuppliers`,
`statsbestmanufacturers`, `statsbestvouchers`, `statscheckup`, `statsforecast`, `dashproducts` - so
those need their own changes to stay consistent with core.
